Roster col sort

ronny2205 opened the pull request:
Adding custom sort to the following roster page coloumns:
Program Assigned, Level of Need, Fluency

kevinrobinson commented:
@ronny2205 Awesome, thanks so much for the pull request! :)

kevinrobinson commented on config/database.yml:
```diff
@@ -6,6 +6,8 @@ development:
   pool: 5
   database: student_insights_development
   host: localhost
+  username: postgres
```
We'll need to remove this to merge, but if you can add a Vagrantfile that works with this (or update the database.yml to take an environment variable for Vagrant builds) in a separate PR that'd be most welcome!

kevinrobinson commented on app/assets/javascripts/roster/sorts/fluency_sort.js:
```diff
@@ -0,0 +1,25 @@
+    return $.trim(i);
+  }
+
+  var fluency_levels = ["FLEP-Transitioning", "FLEP", "Fluent"];
```
:+1: great way to express this!

kevinrobinson commented on app/assets/javascripts/roster/sorts/fluency_sort.js:
```diff
@@ -0,0 +1,25 @@
+// Extend TableSort with Fluency levels
+(function(){
+
+  function clean_cell(i) {
```
Could you remove the `clean_cell`, it looks like it's not doing anything other than calling `$.trim` in each case?

kevinrobinson commented on app/views/homerooms/show.html.erb:
```diff
@@ -74,19 +74,19 @@
               <span class="table-header">Risk</span>
             </th>
-            <th class="program">
+            <th class="program" data-sort-method="prog_assign_sort">
```
@ronny2205 Check out my [comments on the issue](https://github.com/studentinsights/studentinsights/issues/155#issuecomment-199256848) about this.  While this works I think it's hard to discover what's going on here.  We should do something to help clarify what this is, even if it's just the suggestion to add a comment about what's going on.

kevinrobinson commented:
@ronny2205 Thanks!  If you can address the few minor changes here, it'd be awesome and we can get this merged.  I really appreciate your help :)

erose commented on app/assets/javascripts/roster/sorts/fluency_sort.js:
```diff
@@ -0,0 +1,25 @@
+
+  Tablesort.extend('fluency_sort', function(item) {
+    cleaned_cell = clean_cell(item);
+    fluency_levels.indexOf(cleaned_cell) > 0;
```
Add a 'return' here -- this function won't validate as-is.

ronny2205 commented:
Thanks for the comments, I'll fix it soon.

ronny2205 commented:
@kevinrobinson, 
I made all the changes: 
I removed the clean_cell function, but left the $.trim. Do you think we should get rid of it as well?
Also, let me know if the comments I added are clear enough or if there anything else to change.
Thanks!

kevinrobinson commented:
Awesome! I didn't get to look at this yet but thank you, and hopefully will
be able to shortly so we can get this shipped :)
On Thu, Apr 7, 2016 at 6:05 PM Ronny Almog <notifications@github.com> wrote:

> @kevinrobinson <https://github.com/kevinrobinson>,
> I made all the changes:
> I removed the clean_cell function, but left the $.trim. Do you think we
> should get rid of it as well?
> Also, let me know if the comments I added are clear enough or if there
> anything else to change.
> Thanks!
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/studentinsights/studentinsights/pull/252#issuecomment-207109944>
>

kevinrobinson commented on app/views/homerooms/show.html.erb:
```diff
@@ -74,19 +74,22 @@
               <span class="table-header">Risk</span>
             </th>
-            <th class="program">
+            <!-- The data-sort-method attribute forces Tablesort to use a specific sorting method, instead of the default sort it uses. You can see the method at: app/assets/javascripts/roster/sorts/prog_assign_sort.js -->
```
Nice, thanks for adding this in!  Using HTML comments will send this text all the way down to end-users in the browser though, so comments in ERB files are better expressed like this:

```
<%
 # The data-sort-method attribute forces Tablesort to use a specific sorting method, instead of the default sort it uses.
 # You can see the method at: app/assets/javascripts/roster/sorts/prog_assign_sort.js
%>
```

So that it's only visible to us developers. :)

ronny2205 replied:
Thanks for the correction. Just fixed it and pushed.
Note the there are previous comments in that page, written in HTML format. 
Shall I change them as well?

alexsoble commented:
Hi @ronny2205!

Thanks for this work. I am merging it in, but using a different PR: #422.

Why a different branch / PR? 

I did some light editing to this history to make it a bit more readable and easier for a reviewer to look over the commit messages and understand what's going on. I used Git rebase to do this. If you're not familiar with Git rebase, [here's a great introduction](https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history).

I wrote a more thorough explanation of these changes in #422, along with some screenshots to show folks visually how your changes make the roster sort better. 

Thank you again for contributing! The way you kept updating the work in response to @kevinrobinson's ideas and suggestions is really excellent. 

And let me know if you are going to Code for Boston tonight! I'll be there.

alexsoble commented:
^^ Code for Boston is tomorrow night @ronny2205, not tonight. I'm just overly excited and can't wait.

alexsoble commented:
Merged in this work via #422.

ronny2205 commented:
Thanks @alexsoble for all the comments. Looking forward to working on more issues, and I will probably make it tonight!
